src/nvim/globals.h

```diff
@@ -353,9 +353,11 @@ EXTERN int t_colors INIT(= 256);                // int value of T_CCO
 // position.  Search_match_lines is the number of lines after the match (0 for
 // a match within one line), search_match_endcol the column number of the
 // character just after the match in the last line.
-EXTERN int highlight_match INIT(= false);       // show search match pos
-EXTERN linenr_T search_match_lines;             // lines of of matched string
-EXTERN colnr_T search_match_endcol;             // col nr of match end
+EXTERN int highlight_match INIT(= false);          // show search match pos
```

Comment:
in general, if the variable is set to true/false only, use bool always. If it is set to `MAYBE`,  use TriState. If it's set to anything else, keep int and use integer values directly even if Vim uses true,false for 1,0.